flybywiresim/a32nx
# fix: regressions for flight plan drawing throttling

Fixes #3379
Fixes #3380 
Fixes #3397

## Summary of Changes
- Recompute distance and time on flight plan page periodically.
- Disable smoothing since it's buggy with higher refresh intervals.
- Reintroduce "tryAutoActivateApproach" as this seems needed for fplan to switch to approach waypoints.
- Update flight plan version when setting next waypoint.
- Properly load approach initially from MSFS.

## Screenshots (if necessary)
https://photos.app.goo.gl/DQtBviCVNQAEAGKa7

## References
n/a

## Additional context
Discord username (if different from GitHub): JoJo#7105

## Testing instructions
- Fly a flight plan, ensure that the ND functions correctly and that the flight plan line has a correct refresh rate.
- Ensure no other regressions.
- Go to FPLAN page and ensure ETA/Distance is updating.
- Fly a STAR with approach waypoints.

Note: When popping out the ND display (Alt Gr + Mouse Click), refresh rate seems to be unreasonably low. That is a separate issue that needs to be investigated, but is not related to this PR.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
